app/controllers/projects/samples/metadata/file_imports_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Projects
+  module Samples
+    module Metadata
+      # Controller actions for Project Samples Metadata File Import Controller
+      class FileImportsController < Projects::ApplicationController
+        respond_to :turbo_stream
+
+        def create
+          authorize! @project, to: :update_sample?
+          @imported_metadata = ::Samples::Metadata::FileImportService.new(@project, current_user,
+                                                                          file_import_params).execute
+          if @project.errors.empty?
+            render status: :ok, locals: { type: :success, message: t('.success') }
+          elsif @project.errors.include?(:sample)
+            errors = @project.errors.messages_for(:sample)
+            render status: :partial_content, locals: { type: :alert, message: t('.error'), errors: }
+          else
+            error = @project.errors.full_messages_for(:base).first
```

Comment:
Since we can have multiple errors attached we don't want to just get the first one. I had recently just updated the `transfers_controller` in a similar way. Take a look at the `create` method in `app/controllers/projects/samples/transfers_controller.rb`

Reply:
I can do that but, I don't think you will have more that one error on base.
There are 2 kinds of errors, file validation errors and those thrown from actually updating the metadata. I believe only the first file validation failure throws a SampleMetadataFileImportError, attaches to base, and returns. Therefore, I don't think it matters if I grab all or the first error for base. 
Only the errors attached to sample could have more that one. If a sample fails updating the metadata, it attaches the error and continues to the next sample.
Actually, I was planning to show base and sample errors differently in my UI wire-frames. 
Can you think of a scenario I might have multiple base errors? I can add a new test for that.

Reply:
I completely misread that, but yeah looking over the service it will only have one error message attached here. Disregard